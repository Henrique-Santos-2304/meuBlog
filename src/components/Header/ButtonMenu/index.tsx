import { Button } from "@chakra-ui/react";
import { DragHandleIcon } from "@chakra-ui/icons";

type ButtonProps = { onOpen?: () => void };

const ButtonMenu = ({ onOpen }: ButtonProps) => (
  <Button
    aria-label="Botão para abrir um menu de opções"
    title="Abra o Menu"
    onClick={onOpen}
    padding="2.1rem 1rem"
    mr="10px"
    bg="transparent"
    _hover={{ color: "white.300" }}
    _focus={{ border: "none" }}
    rounded="full"
  >
    <DragHandleIcon
      transition="all .5s linear"
      color="gray.100"
      width="22px"
      height="22px"
      _hover={{ color: "blackAlpha.800" }}
    />
  </Button>
);

export default ButtonMenu;
