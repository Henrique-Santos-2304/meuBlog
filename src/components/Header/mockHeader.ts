export const mockHeader = {
  dataHeader: {
    alt: "Minha logo",
    title: "Esse sou eu ",
    imageLogo: {
      url: "https://res.cloudinary.com/defnibbpl/image/upload/v1635695141/a61237_orkfde_816831fe5f.png",
    },
    preText: "web",
    proText: "dev",
  },
};
