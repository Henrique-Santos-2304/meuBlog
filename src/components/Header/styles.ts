import styled, { css } from "styled-components";

export const WrapperHeader = styled.header`
  max-width: ${({ theme }) => theme.grid.containerContent};
  height: 12vh;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 3rem;

  @media (min-width: ${({ theme }) => theme.media.medium}) {
    margin-left: 2rem;
  }
`;
export const ContentLogo = styled.div`
  padding: ${({ theme }) => theme.spacings.xxsmall} 1rem;
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const LogoImage = styled.div`
  width: 3.4rem;
  height: 3.6rem;
  border-radius: 50%;
  position: relative;
  margin-right: 0.3rem;
`;
export const Logo = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxbiglarge};
    font-family: ${theme.font.family.Tangerine};
    color: ${theme.colors.effectsPrimary};

    font-weight: 800;
    span {
      color: ${theme.colors.secondary};
    }
  `}
`;
