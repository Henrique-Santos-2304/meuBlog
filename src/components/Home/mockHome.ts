export const mockHome = {
  dataHome: {
    ariaLabel: "Minha Apresentação",
    myDescription: {
      welcome: "Bem vindo",
      name: "Sou Henrique",
      devDescription: "Desenvolvedor Front End",
    },
    photoUser: {
      alt: "esse sou eu ",
      arialabel: "Foto do Desenvolvedor dono do Portfólio",
      userImage: {
        url: "https://res.cloudinary.com/defnibbpl/image/upload/v1635695141/a61237_orkfde_816831fe5f.png",
      },
    },
  },
};
