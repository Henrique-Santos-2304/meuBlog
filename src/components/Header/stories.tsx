import { Story, Meta } from "@storybook/react";
import Header from ".";

export default {
  title: "Header",
  component: Header,
} as Meta;

export const Basic: Story = (args) => <Header {...args} check={false} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <Header {...args} check={false} />;
