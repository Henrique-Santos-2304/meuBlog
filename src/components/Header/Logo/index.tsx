import * as S from "./styles";
import { Box } from "@chakra-ui/react";

const Logo = () => (
  <S.Wrapper>
    <Box padding="4" maxW="3xl">
      <S.PreText aria-label="logo dev">
        Web<span>Dev</span>
      </S.PreText>
    </Box>
  </S.Wrapper>
);

export default Logo;
