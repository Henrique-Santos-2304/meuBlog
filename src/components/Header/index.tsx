/* eslint-disable @next/next/no-img-element */
import Container from "components/globalComponents/Container";
import MenuMobile from "components/globalComponents/MenuNav/MenuMobile";
import * as S from "./styles";
import Image from "next/image";

const Header = () => (
  <Container asHeight="small">
    <MenuMobile />

    <S.WrapperHeader>
      <S.ContentLogo aria-label="Logo do PortfÃ³lio">
        <S.LogoImage>
          <Image
            src="https://res.cloudinary.com/defnibbpl/image/upload/v1635273619/a61237_orkfde.png"
            layout="fill"
          />
        </S.LogoImage>
        <S.Logo>
          Web<span>Dev</span>
        </S.Logo>
      </S.ContentLogo>
    </S.WrapperHeader>
  </Container>
);

export default Header;
