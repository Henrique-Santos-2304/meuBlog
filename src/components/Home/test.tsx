import { renderTheme } from "utils/testRenderTheme";
import { screen } from "@testing-library/react";
import Header from ".";
import { mockHome } from "./mockHome";

describe("Header", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Header {...mockHome} />);
    expect(container).toMatchSnapshot();
  });
  it("should my Description and photo ", () => {
    renderTheme(<Header {...mockHome} />);
    const apresentation = screen.getByTestId(/apresentation/i);
    const name = screen.getByText(/henrique/i);
    const myImage = screen.getByRole("img", {
      name: /Foto do Desenvolvedor dono do Portf√≥lio/i,
    });

    expect(apresentation).toBeInTheDocument();
    expect(name).toBeInTheDocument();
    expect(myImage).toBeInTheDocument();
  });
});
