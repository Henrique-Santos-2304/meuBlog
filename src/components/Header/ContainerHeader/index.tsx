import { Container, Flex, useDisclosure } from "@chakra-ui/react";
import ButtonMenu from "../ButtonMenu";
import Menu from "components/Menu";
import Logo from "../Logo";

const ContainerHeader = () => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <Container
      position="fixed"
      top="0"
      left="0"
      maxW="full"
      p="0"
      bgGradient="linear(to-l,#a8eb12 ,#00bf72 ,#008793 ,#004d7a 45%, #051937)"
    >
      <Container
        p="0"
        maxW="container.md"
        direction="row"
        centerContent
        height="8rem"
      >
        <Flex
          w="100%"
          h="100%"
          alignItems="center"
          px="2rem"
          justifyContent="space-between"
        >
          <Logo />
          <ButtonMenu onOpen={onOpen} />

          <Menu onClose={onClose} isOpen={isOpen} />
        </Flex>
      </Container>
    </Container>
  );
};
export default ContainerHeader;
