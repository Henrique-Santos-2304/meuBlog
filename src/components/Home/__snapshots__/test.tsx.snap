// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render  1`] = `
.c0 {
  max-width: 85rem;
  margin: 0 auto;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  padding: 1rem 2.9rem 1rem 1.4rem;
  margin: 1rem 0 0 2rem;
  background: rgba(255,255,255,0.15);
  color: #1e1e1e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-weight: 500;
  border-radius: 2rem;
}

.c1::before {
  content: "";
  position: absolute;
  top: 50%;
  left: 1rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  height: 80%;
  width: 0.3rem;
  background: #b7f600;
}

.c2 {
  margin-left: 1.3rem;
}

.c2 span {
  color: #b7f600;
}

.c3 {
  text-align: start;
  font-size: 2.3rem;
  margin-left: 1.3rem;
  font-weight: 300;
}

.c3 strong {
  color: #b7f600;
  font-style: italic;
}

.c4 {
  margin-left: 1.3rem;
}

.c5 {
  position: relative;
  width: 30rem;
  height: 41rem;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
}

@media ( min-width:620px) {
  .c0 {
    padding: 0 2.4rem;
  }
}

@media ( min-width:620px) {
  .c0 {
    padding: 0 5.6rem;
  }
}

@media (min-width:768px) {
  .c1 {
    margin-left: 15rem;
  }
}

@media (min-width:768px) {
  .c5 {
    margin-right: 10rem;
  }
}

<div>
  <section
    class="c0"
  >
    <article
      aria-label="Minha Apresentação"
      class="c1"
      data-testid="apresentation"
    >
      <p
        class="c2"
      >
        Bem vindo(a),
      </p>
      <h1
        class="c3"
        title="Meu Nome"
      >
        Sou
        <strong>
           Henrique
        </strong>
        ,
      </h1>
      <p
        class="c4"
      >
        Desenvolvedor FrontEnd
      </p>
    </article>
    <main
      class="c5"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt="Esse sou eu "
          aria-label="Foto do Desenvolvedor dono do Portfólio"
          class=""
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </div>
    </main>
  </section>
</div>
`;
